app/routes/report/probeResults.js

```diff
@@ -36,14 +58,49 @@ router.route('/:target([A-z0-9-]{36})')
     return res.json(req.target.view('public'));
   })
   .put(async (req, res) => {
-    // Update db entry
+    // Validate request against schema
+    try {
+      validateAgainstSchema(updateSchema, req.body, false);
+    } catch (error) {
+      const message = `Error while validating probe results update request ${error}`;
+      logger.error(message);
+      return res.status(HTTP_STATUS.BAD_REQUEST).json({error: {message}});
+    }
+
+    // If a new gene is provided, search for the gene
+    // and if found use it to replace the old gene
```

Comment:
Can you explain this a bit further? I'm not seeing how `geneId` is being used here

Reply:
GeneId is a foreign key that references a specific gene in the gene table. It is used to eager load gene data when retrieving probe results. 

The change below this comment allows swapping the gene that is linked to this probe result. The swap is done by providing a gene ident and checking that the gene exists. If it does, the geneId foreign key is swapped from the old genes id to the new genes id. This change is then committed to the database and the model is reloaded with the updated data and the new eager loaded gene data before being returned.